### PR TITLE
[Watch app] opening order shows yellow warning icon

### LINKED ISSUE #14127: [Watch app] opening order shows yellow warning icon
**Describe the bug**

When opening an order in the watchOS app the order details are shown for a split second, after that the yellow warning icon is shown.

**To Reproduce**
1. Open Woo watchOS app
2. Scroll to order list
3. Tap on any of the orders in the list
4. See the warning icon

**Screenshots**

https://github.com/user-attachments/assets/d389127e-ff2f-4549-a241-debf62440145

**Expected behavior**

Order details should show (without that warning icon).

**Mobile Environment**
Please include:
- Device: Apple Watch Series 10
- watchOS version: watchOS 11.0.1 (22R361)
- WooCommerce iOS version: 20.6 (tested also with 20.5)

### PR DESCRIPTION
**Pull request by bozidarsevo**
Closes: #14127

## Description

When using Woo Watch app and trying to open order details, by tapping order in order list, the app would show a yellow warning icon.

When we run the app from Xcode the console shows this messages:

```
Do not put a navigation destination modifier inside a "lazy” container, like `List` or `LazyVStack`. These containers create child views only when needed to render on screen. Add the navigation destination modifier outside these containers so that the navigation stack can always see the destination. There's a misplaced `navigationDestination(for:destination:)` modifier for type `Order`. It will be ignored.
A NavigationLink is presenting a value of type “Order” but there is no matching navigationDestination declaration visible from the location of the link. The link cannot be activated.

Note: Links search for destinations in any surrounding NavigationStack, then within the same column of a NavigationSplitView.
```

After a debugging session I changed the usage of `NavigationLink` to use `NavigationLink(destination: ...)` like this:
```
NavigationLink(destination: OrderDetailView(order: order)) {
  OrderListCard(order: order)
}
```
With that change app works normally.

## Steps to reproduce

- Install Woo from App Store, Testflight or run from `trunk`
- Open Woo app on Apple Watch
- Tap on one of the orders in the list
- You should either see a yellow warning icon or tapping one of the orders will not even do that (you will stay in the order list)

## Testing information

- Open Woo app on Apple Watch
- Tap on one of the orders in the list
- Order details for that order should show
- Scroll order details to see if all info is shown

### Trunk:

https://github.com/user-attachments/assets/a0ecc26f-dd62-4640-a81a-f6b26c8ae755

### Testflight (21.0.0):

https://github.com/user-attachments/assets/b1a826a3-64cd-4863-95bf-5c39476ce3e9 

### This PR:

https://github.com/user-attachments/assets/f16f9c4c-08f0-4aaf-8401-33d17f133215

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@
 -----
 
 - [internal] Updated Xcode version to 16.1 [https://github.com/woocommerce/woocommerce-ios/pull/14225]
+- [**] Fixed: properly open and show order details in Watch app
```
Comment (joshheald) [inline RELEASE-NOTES.txt]:
```suggestion
- [**] Fixed: properly open and show order details in Watch app [https://github.com/woocommerce/woocommerce-ios/pull/14306]
```

File: WooCommerce/Woo Watch App/Orders/OrdersListView.swift [hunk 2]
```diff
@@ -114,17 +114,14 @@ struct OrdersListView: View {
     /// Data: List with live order content.
     ///
     @ViewBuilder private func dataView(orders: [Order]) -> some View {
-        List() {
+        List {
             ForEach(orders, id: \.number) { order in
-                NavigationLink(value: order) {
+                NavigationLink(destination: OrderDetailView(order: order)) {
```
Comment (joshheald) [inline WooCommerce/Woo Watch App/Orders/OrdersListView.swift]: The non-closure version of `NavigationLink` is deprecated from 17.5, but this version is available instead:

```suggestion
                NavigationLink {
                    OrderDetailView(order: order)
                } label: {
```

**Review by joshheald — APPROVED** [review]
Works well (after I spent an embarrassingly long time figuring out that I needed to target my watch for Xcode to build to, not just assume that running it on my phone would install to the watch as well.)

I've made a couple of inline suggestions.

Alternatively, you could could keep the `navigationDestination(for:)` style of navigation by moving that modifier out of the order list entirely, so the `navigationDestination` modifier is always installed on the view and will work correctly when the value is set by the `NavigationLink`. 

It's not a List or LazyVStack that's at fault in this case; I found that it had to be moved to the `TabView` in `WooApp`. It looks like the tabs are created lazily, so the `navigationDestination` modifier needs to be at quite a high level to do it that way.

**Pull request merged**